### DISCUSSION TITLE
Error handling while executing the model server commands

### DISCUSSION THREAD

**Discussion by shinesamjacob** [eclipse-emfcloud/emfcloud#210] [ModelServer]
We have written our model server command, which we execute from the front end. In specific scenarios, we need to validate and throw the exception/messages to the frontend. The version of the emf cloud is 0.7.0 and uses the v1 client.

Is there any way to handle this scenario without writing the custom routing endpoints?

**Comment by ndoschek** [comment]
Hi @shinesamjacob,

If it is any EMF command or JSON patch command (with v2 possbile) you can interpret the the Model Server's response on the frontend and show the results for example as notificiations in Theia. 

You could have a look at this simple developer example we provide in the `emfcloud-modelserver-theia` project: [`api-test-menu.ts`](https://github.com/eclipse-emfcloud/emfcloud-modelserver-theia/blob/master/examples/dev-example/src/browser/api-test-menu.ts)

For example [the `GetAllCommand` command](https://github.com/eclipse-emfcloud/emfcloud-modelserver-theia/blob/68f54274c961bab65e41495ccce4fcab5d4cd27f/examples/dev-example/src/browser/api-test-menu.ts#LL580C34-L580C47) can be triggered via Menu `Model Server > getAll()`.

Menu entry:
![image](https://github.com/eclipse-emfcloud/emfcloud/assets/47414588/754372fd-9043-4986-a8b2-565ed69be2c9)

This executes the `getAll` command (which queries all loaded models from the Model Server) and once the response is received, we call `printResponse` which shows the result as notification message in Theia.

Notification:
![image](https://github.com/eclipse-emfcloud/emfcloud/assets/47414588/1ec36b66-f30c-4c33-894a-42dd5e72855c)

Result:
![image](https://github.com/eclipse-emfcloud/emfcloud/assets/47414588/2d835e06-d4de-42b8-be14-de94308afd0f)

To start the example, [please check the main README](https://github.com/eclipse-emfcloud/emfcloud-modelserver-theia/tree/master).

HTH and best regards,
Nina

**Comment by eneufeld** [comment]
Please reopen if issue still persists
